farcy/helpers.py

```diff
@@ -184,107 +196,99 @@ class Config(object):
 
     """Holds configuration for Farcy."""
 
-    def __init__(self, repository=None):
-        """Initialize a config with default values."""
-        self.repository = repository
-        self.start_event = None
-        self.debug = False
-        self.exclude_paths = []
-        self.limit_users = None
-        self.log_level = 'NOTSET'
-        self.pr_issue_report_limit = 128
-        self.session = None
-
-        self._dirty = set()
-        self._config_path = os.path.join(CONFIG_DIR, 'farcy.conf')
-
-    def ensure_session(self):
-        """Load or create GitHub session."""
-        if self.session is not None:
-            return
-        self.session = get_session()
-
-    def load_config_file(self):
-        """
-        Load default values from configuration values.
-
-        Will not overwrite any value if already set.
-        """
-        if not os.path.isfile(self._config_path):
-            return
-
-        allowed = ('start_event', 'debug', 'exclude_paths', 'limit_users',
-                   'log_level', 'pr_issue_report_limit')
-
-        config_file = ConfigParser()
-        config_file.read(self._config_path)
-        if self.repository is None and config_file.has_option(
-                'default', 'repository'):
-            self.repository = config_file.get('default', 'repository')
-
-        for section in (self.repository, 'default'):
```

Comment:
The new approach introduces a regression: it no longer allows users to specify values in the default section that will be used if the value is not specified for the repo.

I'm fine with that, just wanted to point it out.